src/ecdsa/ot_based_ecdsa/test.rs

```diff
@@ -327,3 +327,132 @@ fn test_e2e_random_identifiers_with_rerandomization() -> Result<(), Box<dyn Erro
     run_sign_with_rerandomization(presign_result, public_key.to_element(), msg)?;
     Ok(())
 }
+
+fn split_even_odd<T: Clone>(v: Vec<T>) -> (Vec<T>, Vec<T>) {
+    let mut even = Vec::with_capacity(v.len() / 2 + 1);
+    let mut odd = Vec::with_capacity(v.len() / 2);
+    for (i, x) in v.iter().enumerate() {
+        if i % 2 == 0 {
+            even.push(x.clone());
+        } else {
+            odd.push(x.clone());
+        }
+    }
+    (even, odd)
+}
```

Comment:
Nit: While this is valid - you can also use the `partition` method on iterators for this:

```suggestion
fn split_even_odd<T: Clone>(v: Vec<T>) -> (Vec<T>, Vec<T>) {
    v.into_iter().enumerate().partition(|(i, _)| i % 2 == 0).map(|(_, x)| x)
}
```

Reply:
Right yeah my suggestion is wrong. Sure though this suggestion while a bit cleaner is less efficient since it requires two passes of the same data.

Reply:
Haha I do not think any of the two options is actually better in terms of complexity.
I will keep it as is.